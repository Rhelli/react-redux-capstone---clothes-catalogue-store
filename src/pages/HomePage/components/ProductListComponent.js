/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import PropTypes from 'prop-types';
import ProductListImageComponent from './ProductListImageComponent';
import ProductListInfoComponent from './ProductListInfoComponent';

const ProductListComponent = props => {
  const {
    productName, price, images, itemClickThrough,
  } = props;

  return (
    <div onClick={itemClickThrough}>
      <ProductListImageComponent images={images} />
      <ProductListInfoComponent
        productName={productName}
        price={price}
      />
    </div>
  );
};

ProductListComponent.propTypes = {
  productName: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  images: PropTypes.string.isRequired,
  itemClickThrough: PropTypes.func.isRequired,
};

export default ProductListComponent;
