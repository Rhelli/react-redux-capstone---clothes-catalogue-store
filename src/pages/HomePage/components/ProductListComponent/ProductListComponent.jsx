import React from 'react';
import PropTypes from 'prop-types';
import ProductListImageComponent from '../ProductListImageComponent/ProductListImageComponent';
import ProductListInfoComponent from '../ProductListInfoComponent/ProductListInfoComponent';
import styles from './ProductListComponent.module.scss';

const ProductListComponent = props => {
  const {
    productName, price, images, itemClickThrough,
  } = props;

  return (
    <div className={styles.productListItem} onClick={itemClickThrough} onKeyUp={itemClickThrough} role="button" tabIndex="-1">
      <ProductListImageComponent images={images} />
      <ProductListInfoComponent
        productName={productName}
        price={price}
      />
    </div>
  );
};

ProductListComponent.propTypes = {
  productName: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  images: PropTypes.string.isRequired,
  itemClickThrough: PropTypes.func.isRequired,
};

export default ProductListComponent;
