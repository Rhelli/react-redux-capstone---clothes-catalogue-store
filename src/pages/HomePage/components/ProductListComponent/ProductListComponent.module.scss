.productListItem {
  width: 48%;
  background-color: #FBEEE6;
  border-radius: 15px;
  margin-bottom: 10px;
}
