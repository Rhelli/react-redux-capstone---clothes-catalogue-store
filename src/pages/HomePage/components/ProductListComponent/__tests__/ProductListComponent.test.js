import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProductListComponent from '../ProductListComponent';

configure({ adapter: new Adapter() });

const productName = 'Black T-Shirt';
const price = '10.00';
const images = 'thisIsAnImageURL';
const itemClickThrough = () => 'This is a mock function';

const productListComponent = <ProductListComponent productName={productName} price={price} images={images} itemClickThrough={itemClickThrough} />;

describe('The appearance of the ProductListComponent', () => {
  it('Matches the ProductListComponent snapshot', () => {
    const tree = renderer.create(productListComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has one single child container, encapsulating both productList components', () => {
    const { container } = render(productListComponent);
    expect(container.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.classList.contains('productListItem')).toBe(true);
  })
});
