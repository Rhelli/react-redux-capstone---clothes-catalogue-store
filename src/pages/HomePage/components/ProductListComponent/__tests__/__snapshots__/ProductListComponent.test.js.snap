// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance of the ProductListComponent Matches the ProductListComponent snapshot 1`] = `
<div
  className="productListItem"
  onClick={[Function]}
>
  <div
    className="productListImage"
  >
    <img
      alt=""
      src="thisIsAnImageURL"
    />
  </div>
  <div
    className="productListInfoContainer"
  >
    <div
      className="productListInfoNameContainer"
    >
      <h3>
        Black T-Shirt
      </h3>
    </div>
    <div
      className="productListInfoPriceContainer"
    >
      <h5>
        Â£
        10.00
      </h5>
    </div>
  </div>
</div>
`;
